examples/structured_2d_dgsem/elixir_shallowwater_well_balanced_wet_dry.jl

```diff
@@ -0,0 +1,196 @@
+
+stepsize_callback = StepsizeCallback(cfl=1.0)
+
+callbacks = CallbackSet(summary_callback, analysis_callback, alive_callback, save_solution, stepsize_callback)
+
+stage_limiter! = PositivityPreservingLimiterShallowWater(variables=(Trixi.waterheight,))
+
+###############################################################################
+# run the simulation
+
+sol = solve(ode, SSPRK43(stage_limiter!); dt=1.0,
+            ode_default_options()..., callback=callbacks, adaptive=false);
+
+summary_callback() # print the timer summary
+
+###############################################################################
+# Workaround to compute the well-balancedness error for this partiular problem
+# that has two reference water heights. One for a lake to the left of the
+# discontinuous bottom topography `H0_upper = 2.5` and another for a lake to the
+# right of the discontinuous bottom topography `H0_lower = 1.5`.
+
+# Declare a special version of the function to compute the lake-at-rest error
+# OBS! The reference water height values are hardcoded for convenience.
+function lake_at_rest_error_two_level(u, x, equations::ShallowWaterEquations2D)
+    h, _, _, b = u
+
+  # For well-balancedness testing with possible wet/dry regions the reference
+  # water height `H0` accounts for the possibility that the bottom topography
+  # can emerge out of the water as well as for the threshold offset to avoid
+  # division by a "hard" zero water heights as well.
+  if x[1] < 0.5
+    H0_wet_dry = max( 2.5 , b + equations.threshold_limiter )
+  else
+    H0_wet_dry = max( 1.5 , b + equations.threshold_limiter )
+  end
+
+  return abs(H0_wet_dry - (h + b))
+end
+
+# point to the data we want to analyze
+u = Trixi.wrap_array(sol[end], semi)
+# Perform the actual integration of the well-balancedness error over the domain
+l1_well_balance_error = Trixi.integrate_via_indices(u, mesh, equations, semi.solver, semi.cache; normalize=true) do u, i, j, element, equations, solver
+  x_node = Trixi.get_node_coords(semi.cache.elements.node_coordinates, equations, solver, i, j, element)
+  # We know that the discontinuity is a vertical line. Slightly augment the x value by a factor
+  # of unit roundoff to avoid the repeted value from the LGL nodes at at interface.
+  if i == 1
+    x_node = SVector(nextfloat(x_node[1]) , x_node[2])
+  elseif i == nnodes(semi.solver)
+    x_node = SVector(prevfloat(x_node[1]) , x_node[2])
+  end
+  u_local = Trixi.get_node_vars(u, equations, solver, i, j, element)
+  return lake_at_rest_error_two_level(u_local, x_node, equations)
+end
+
+# report the well-balancedness lake-at-rest error to the screen
+println("─"^100)
+println(" Lake-at-rest error for '", Trixi.get_name(equations), "' with ", Trixi.summary(solver),
+                  " at final time " * Trixi.@sprintf("%10.8e", tspan[end]))
+
+Trixi.@printf(" %-12s:", Trixi.pretty_form_utf(lake_at_rest_error))
+Trixi.@printf("  % 10.8e", l1_well_balance_error)
```

Comment:
```suggestion
println(" Lake-at-rest error for '", Trixi.get_name(equations), "' with ", summary(solver),
                  " at final time " * @sprintf("%10.8e", tspan[end]))

@printf(" %-12s:", Trixi.pretty_form_utf(lake_at_rest_error))
@printf("  % 10.8e", l1_well_balance_error)
```
To clarify that we're not doing anything strange with these standard macros/functions